src/almanack/metrics/data.py

```diff
@@ -1182,3 +1195,71 @@ def find_doi_citation_data(repo: pygit2.Repository) -> Dict[str, Any]:
                 LOGGER.warning(f"Error during OpenAlex exact DOI lookup: {e}")
 
     return result
+
+
+def compute_sustainability_score(
+    table: List[Dict[str, Union[int, float, bool]]]
+) -> float:
+    """
+    Computes a sustainability score by normalizing
+    numeric metrics and incorporating boolean metrics.
+    Adjusts for metrics that should be inversely
+    proportional to the sustainability score.
+
+    Args:
+        data (List[Dict[str, Union[int, float, bool]]]):
+            A list of dictionaries containing metrics.
+            Each dictionary must have a "result" key
+            with a value that is an int, float, or bool.
+              Optionally, a "direction" key may be
+            included for numeric values to specify the
+            relationship to sustainability:
+            - 1 (positive correlation)
+            - 0 (no correlation)
+            - -1 (negative correlation)
+
+    Returns:
+        float:
+            The computed sustainability score, normalized between 0 and 1.
+    """
+    results = []
+    bool_results = []
+
+    # Gather numeric and boolean values with a direction
+    for item in table:
+        if isinstance(item["result"], bool) and item["direction"] != 0:
+            # for direction == 1 we treat True as positive correction
+            # and False as a negative correlation.
+            if item["direction"] == 1:
+                bool_results.append(1 if item["result"] else 0)
+            # for direction == -1 we treat True as negative correction
+            # and False as a positive correlation.
+            elif item["direction"] == -1:
+                bool_results.append(0 if item["result"] else 1)
+
+        elif isinstance(item["result"], (int, float)) and item["direction"] != 0:
+            # Numeric values inverted or kept as-is based on direction
+            results.append(
+                item["result"] if item["direction"] == 1 else 1 - item["result"]
+            )
+
+    # Normalize numeric values if any
+    if results:
+        scaler = RobustScaler()
+        normalized_results = scaler.fit_transform(
```

Comment:
Are you scaling all numeric values of the same repository together to form a single score? If so, I'm confused as to why? Can you provide an example of a numeric score? Is entropy an example? Do all of the numeric scores have the same importance to sustainability? Are they equivalent to the non-numeric indicators? (What is your evidence?)

Reply:
Thanks for checking on this - I tried to remain neutral on the aspect of importance per metric, instead attempting to scale the values for any directional value (i.e. not just meta). Pondering this as we've developed the project made me feel that weighting of the metrics individually could be challenging as the success of each project varies based on the people, technology, and goals. The current attempt here tries to measure the scaled metrics together by only assessing whether they're "additive" (direction 1), or detractive (direction -1) to create a sustainability score. I'll give this some more thought and follow up.

Reply:
I couldn't find a clear method by which we could quantify the numeric amounts as a composite and as a result removed the use of numeric results entirely (now only using the bools for a simpler calculation). I feel that this will align with future efforts where we need to guide on whether a particular aspect is a "good" or "bad" practice.